hudi-utilities/src/test/java/org/apache/hudi/utilities/deltastreamer/TestHoodieDeltaStreamer.java

```diff
@@ -3376,6 +3376,45 @@ void testErrorTableSourcePersist(WriteOperationType writeOperationType, boolean
     assertRecordCount(950, tableBasePath, sqlContext);
   }
 
+  @ParameterizedTest
+  @EnumSource(HoodieTableType.class)
+  void testDeltaSyncWithAutoKeyGenAndImmutableOperations(HoodieTableType tableType) throws Exception {
```

Comment:
can we try a mix of bulk insert, insert and upsert. 
even though we might set upsert, for auto key gen, hudi should automatically switch it to insert. 

lets validate that from commit metadata as well.